### PR TITLE
fix: make pkce code_verifier spec compliant #125

### PR DESCRIPTION
**Pull request by moximoti**
follow recommendations for code_verifier: https://datatracker.ietf.org/doc/html/rfc7636#section-4.1

### CODE REVIEW

**Comment by fforootd** [comment]
Hi @moximoti thanks for the PR. Thats a good catch 🥇 

@livio-a IMO this looks fine, what do you think?

File: pkg/client/rp/relaying_party.go [hunk 1]
```diff
@@ -288,7 +289,7 @@ func AuthURLHandler(stateFn func() string, rp RelyingParty) http.HandlerFunc {
 
 //GenerateAndStoreCodeChallenge generates a PKCE code challenge and stores its verifier into a secure cookie
 func GenerateAndStoreCodeChallenge(w http.ResponseWriter, rp RelyingParty) (string, error) {
-	codeVerifier := uuid.New().String()
+	codeVerifier := base64.URLEncoding.EncodeToString([]byte(uuid.New().String()))
```
[the remark this answers is not in the document]
Comment (livio-a) [reply pkg/client/rp/relaying_party.go:292]: the encoded value is passed as query param so i suggest using no padding (`=`)

**Review by livio-a — COMMENTED** [review]
Hi @moximoti 

Thanks for the PR and sorry for the late response, I've been pretty busy last week.

Looks pretty good. Just see my comment concerning padding.

**Comment by moximoti** [comment]
sure, no padding definitely makes sense. especially if you want to be able to change uuid for something else in the future. 
(although in this case, due to the fact that UUID is always 36 chars and therefore a multiple of 4 the result of both encodings do not include padding characters)

**Pull request merged**